SU2_CFD/src/solvers/CMeshSolver.cpp

```diff
@@ -499,33 +498,37 @@ void CMeshSolver::DeformMesh(CGeometry **geometry, CNumerics **numerics, CConfig
   /*--- Solve the linear system. ---*/
   Solve_System(geometry[MESH_0], config);
 
+  SU2_OMP_PARALLEL {
+
   /*--- Update the grid coordinates and cell volumes using the solution
      of the linear system (usol contains the x, y, z displacements). ---*/
   UpdateGridCoord(geometry[MESH_0], config);
 
   /*--- Update the dual grid. ---*/
   UpdateDualGrid(geometry[MESH_0], config);
 
-  /*--- Check for failed deformation (negative volumes). ---*/
-  /*--- This is not recorded as it does not influence the solution. ---*/
-  AD::StopRecording();
-  SetMinMaxVolume(geometry[MESH_0], config, true);
-  if (ActiveTape) AD::StartRecording();
-
   /*--- The Grid Velocity is only computed if the problem is time domain ---*/
   if (time_domain) ComputeGridVelocity(geometry[MESH_0], config);
 
   /*--- Update the multigrid structure. ---*/
   UpdateMultiGrid(geometry, config);
 
+  } // end parallel
+
+  /*--- Check for failed deformation (negative volumes). ---*/
+  /*--- This is not recorded as it does not influence the solution. ---*/
+  AD::StopRecording();
```

Comment:
You can use AD_BEGIN_PASSIVE/AD_END_PASSIVE have passive regions. Although I am not a big fan of preprocessor macros and maybe we find another way soon, but at least it is consistent with the rest for now :)

Reply:
Yeah I thought of making it a function of the AD namespace, maybe I'll do it now :)

Reply:
Made a function and replaced both this kind of logic and the macro.